public/js/components/AtomEdit/CustomEditors/StoryQuestionFields/Question.js

```diff
@@ -27,21 +39,43 @@ export class StoryQuestionsQuestion extends React.Component {
     this.props.onUpdateField(questionWithId);
   }
 
-  render () {
-    const answersCount = (this.props.fieldValue && this.props.fieldValue.answers) ? this.props.fieldValue.answers.length : 0;
+  onSelect = (snippet) => {
+    this.setState((oldState) => Object.assign({}, oldState, { answers: oldState.answers.concat({
+      title: snippet.title,
+      answerId: `atom/${snippet.atomType}/${snippet.id}`, 
+      answerType: 'ATOM' 
+    })}));
+  }
 
+  onDelete = (i) => {
```

Comment:
I find it generally safer to rely on an `answerId` instead of an index when deleting stuff. The user might click the button twice quickly*, and then you might receive two delete events with the same index. The code would then delete two items even though the user only wanted to delete one.

\* _Theoritically, this shouldn't happen unless there are worker threads, but I don't know if React uses / will use them._

Reply:
Yeah there are probably UX pitfalls all over the place. I need to iron out an elegant way to polish that, hopefully without muddying the logic of my components. This kind of thing should be handled at a lower level and not surface in your business logic. (no worker thread)